src/hotspot/share/memory/classLoaderMetaspace.hpp

```diff
@@ -75,8 +78,14 @@ class ClassLoaderMetaspace : public CHeapObj<mtClass> {
   metaspace::MetaspaceArena* non_class_space_arena() const   { return _non_class_space_arena; }
   metaspace::MetaspaceArena* class_space_arena() const       { return _class_space_arena; }
 
-public:
+  bool have_class_space_arena() const { return _class_space_arena != nullptr; }
```

Comment:
I'd prefer not to. 

This logic (when -UCCP class space arena is NULL, with the implicit assumption that both are different entities) has been in there forever, and changing that is out of scope for and unrelated to this PR. I am not sure what will break if I change this but don't want to chase risk test errors at this point (one example, reporting would have to be adapted to recognize that both arenas are the same, and there are plenty of tests that would also need to be fixd).

This can be done in a follow-up RFE if necessary.

Reply:
OK, that's fine.